RELEASES.md

```diff
@@ -0,0 +1,414 @@
+
+`EnvHandler` is no longer exposed to the user and instead the environment is now always accessed via
+`self.env()`.
+
+<table style="width: 100%;">
+<tr>
+<th>Before</th>
+<th>After</th>
+</tr>
+<tr>
+<td>
+
+**Getting the caller:**
+
+```rust
+let caller = env.caller();
+```
+
+**Emitting an event:**
+
+```rust
+env.emit(...)
+```
+
+</td>
+<td>
+
+**Getting the caller:**
+
+```rust
+let caller = self.env().caller();
+```
+
+**Emitting an event:**
+
+```rust
+self.env().emit_event(...)
+```
+
+</td>
+</tr>
+</table>
+
+> Note: The name of the function used to emit an event was updated to `emit_event`.
+
+## Message Functions
+
+We used to use `pub(external)` to tag functions that could be called by the outside world.
+
+We now simply add the attribute `#[ink(message)]`.
+
+<table style="width: 100%;">
+<tr>
+<th>Before</th>
+<th>After</th>
+</tr>
+<tr>
+<td>
+
+```rust
+pub(external) fn total_supply(&self) -> Balance {
+    *self.total_supply
+}
+```
+
+</td>
+<td>
+
+```rust
+#[ink(message)]
+fn total_supply(&self) -> Balance {
+    *self.total_supply
+}
+```
+
+</td>
+</tr>
+</table>
+
+## Defining a Constructor
+
+We used to define our constructor by implementing the `Deploy` trait and defining the `deploy`
+function.
+
+But now our constructor function is in the same place as the rest of our contract functions, within
+the general implementation of the storage struct.
+
+We tag these functions with the `#[ink(constructor)]` attribute. We can create multiple different
+constructors by simply creating more functions with the same tag. You can name a constructor
+function whatever you want (except starting with `__ink` which is reserved for all functions).
+
+<table style="width: 100%;">
+<tr>
+<th>Before</th>
+<th>After</th>
+</tr>
+<tr>
+<td>
+
+```rust
+impl Deploy for Erc20 {
+    fn deploy(&mut self, init_supply: Balance) {
+        let caller = env.caller();
+        self.total_supply.set(init_value);
+        self.balances.insert(caller, init_supply);
+        env.emit(Transfer {
+            from: None,
+            to: Some(env.caller()),
+            value: init_value
+        });
+    }
+}
+```
+
+</td>
+<td>
+
+```rust
+impl Erc20 {
+    #[ink(constructor)]
+    fn new(&mut self, initial_supply: Balance) {
+        let caller = self.env().caller();
+        self.total_supply.set(initial_supply);
+        self.balances.insert(caller, initial_supply);
+        self.env().emit_event(Transferred {
+            from: None,
+            to: Some(caller),
+            amount: initial_supply,
+        });
+    }
+}
+```
+
+</td>
+</tr>
+</table>
+
+## Cross Contract Calls
+
+It is now possible to call ink! messages and ink! constructors. So ink! constructors allow
+delegation and ink! messages can easily call other ink! messages.
+
+Given another ink! contract like `mod Accumulator { ... }`, we can call any of its functions just
+like we would expect:
+
+```rust
+use accumulator::Accumulator;
+let result = Accumulator::my_function()
```

Comment:
This is not really correct syntax unless `my_function` was a `my_constructor` but then again this is only possible in the new ink! 2.0.